lua/bufferline/tabpages.lua

```diff
@@ -30,13 +30,24 @@ local function render(tabpage, is_active, style, highlights)
   local separator_hl = is_active and h.tab_separator_selected.hl_group or h.tab_separator.hl_group
   local chars = constants.sep_chars[style] or constants.sep_chars.thin
   local separator_component = chars[2]
-  local name = padding .. padding .. tabpage.tabnr .. padding
+  local name = padding .. padding .. (tabpage.variables.name or tabpage.tabnr) .. padding
   return {
     { highlight = hl, text = name, attr = { prefix = tab_click_component(tabpage.tabnr) } },
     { highlight = separator_hl, text = separator_component },
   }
 end
 
+function M.rename_tab(tabnr, name)
+  if tabnr == 0 then
+    tabnr = vim.fn.tabpagenr()
+  end
+  if name == "" then
+    name = ""..tabnr..""
```

Comment:
Seems you handle defaulting to `tabnr` above, so instead we can set the variable to `nil`?
```suggestion
    name = nil
```